x-pack/legacy/plugins/ml/public/timeseriesexplorer/timeseries_search_service.ts

```diff
@@ -4,44 +4,59 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-
-
 import _ from 'lodash';
 
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 import { ml } from '../services/ml_api_service';
 import { isModelPlotEnabled } from '../../common/util/job_utils';
+// @ts-ignore
 import { buildConfigFromDetector } from '../util/chart_config_builder';
 import { mlResultsService } from '../services/results_service';
-
-function getMetricData(job, detectorIndex, entityFields, earliestMs, latestMs, interval) {
+import { ModelPlotOutput } from '../services/results_service/result_service_rx';
+import { Job } from '../jobs/new_job/common/job_creator/configs';
+
+function getMetricData(
+  job: Job,
+  detectorIndex: number,
+  entityFields: object[],
+  earliestMs: number,
+  latestMs: number,
+  interval: string
+): Observable<ModelPlotOutput> {
   if (isModelPlotEnabled(job, detectorIndex, entityFields)) {
     // Extract the partition, by, over fields on which to filter.
     const criteriaFields = [];
     const detector = job.analysis_config.detectors[detectorIndex];
     if (_.has(detector, 'partition_field_name')) {
```

Comment:
probably for a future refactor, but there is a lot of lodash in this file which can be completely removed.

Reply:
yeah agree, I just wanted to have a minimum amount of changes and focus on RxJS